### ISSUE TITLE
[micro] Add API to read Reply subject from `micro.Request`

### ISSUE THREAD

**Issue by mcosta74** [nats-io/nats.go#1579]
### Proposed change

To not introduce a breaking change we'd need to add a new interface (`RequestV2` ??) that extends the `Request` and add a new method

```go
type RequestV2 interface {
    Request

    Reply() string
}
```
 

### Use case

In some cases you can trigger a microservice but you don't care about the results so maybe you just send a publish instead of a request.

In this case would be nice to know if a request has a reply subject just to avoid to Marshal a response and try to send back it

```go
func MyHandler(r micro.RequestV2) {
    response := handleRequest()

    if r.Reply() == "" {
        return
    }

   err := r.RespondJSON(response)
   if err != nil {
      // do something
   }
} 
```

### Contribution

I'm interested to help with implementation in case this will be accepted

**Comment by piotrpio** [comment]
Hey @mcosta74. It's a good idea to add the ability to check reply subject, but I'm not sure about introducing a new interface... That would force us to also add v2 of `Handler` interface, along with `Handle` method and `HandlerFunc`.

Another idea would be to add a `RespondOpt` which would allow skipping sending the response if `reply` is empty, but that's not ideal as well since you'd not know whether the response was sent or not.

**Comment by mcosta74** [comment]
Hi @piotrpio , my need is to know if I need to prepare/Marshall the response or I can skip it because there isn't a Reply.

So your 2nd option doesn't fit what I'm looking for.

**Comment by mcosta74** [comment]
> That would force us to also add v2 of Handler interface, along with Handle method and HandlerFunc.

As I said, I'll be happy to implement the change of the proposal is accepted

**Comment by piotrpio** [comment]
@mcosta74, @themartorana Sorry it took so long and thank you for the patience. We discussed it internally and agreed that we will not consider adding public methods to interfaces a breaking change for the client. We have done it in the past with `JetStreamContext` so we can do the same for `micro`. I've created a PR adding a note about that to readme to make it official.

I'll review your PR in a moment, on first glance it looks simple and straightforward enough 😉
